source/simulator/assembly.cc

```diff
@@ -525,7 +513,8 @@ namespace aspect
         Mp_preconditioner_AMG->initialize (system_preconditioner_matrix.block(1,1), Amg_data);
       }
 
-    if (parameters.free_surface_enabled || parameters.include_melt_transport || parameters.use_full_A_block_preconditioner)
+    // TODO only needed for free surface, or for all mesh deformations?
```

Comment:
Probably for all mesh deformations that not just move the whole boundary uniformly. Lets be safe and use it always when mesh deformation is enabled. Remove the TODO